server/src/main/java/org/opensearch/index/engine/EngineConfig.java

```diff
@@ -183,6 +188,9 @@ private static void doValidateCodecSettings(final String codec) {
         switch (codec) {
             case "zstd":
             case "zstd_no_dict":
+                if (FeatureFlags.isEnabled(FeatureFlags.ZSTD_COMPRESSION) == false) {
+                    throw new IllegalArgumentException("Compression level cannot be set before enabling the ZStandard feature.");
```

Comment:
Can we add the exact feature flag name here?

Reply:
done!